src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -2039,6 +2046,34 @@ void Aura::TriggerProcOnEvent(uint8 procEffectMask, AuraApplication* aurApp, Pro
         Remove();
 }
 
+void Aura::HeartbeatResistance(uint32 diff, Unit* caster)
+{
+    Unit* target = GetOwner()->ToUnit();
+
+    if (!target || !caster)
+        return;
+
+    if (m_heartBeatTimer <= diff)
+    {
+        if (target->GetTypeId() == TYPEID_UNIT && caster->GetTypeId() == TYPEID_PLAYER)
+        {
+            SpellSchoolMask schoolMask = m_spellInfo->GetSchoolMask();
+            uint32 resistance = schoolMask != SPELL_SCHOOL_MASK_NORMAL ? target->GetResistance(GetFirstSchoolInMask(schoolMask)) : 0;
+            uint32 breakPct = uint32(resistance / std::powf(target->getLevel(), 1.441f) * 0.10f * 100) + 5;
+
+            if (roll_chance_i(breakPct))
+            {
+                Remove();
+                TC_LOG_DEBUG("spells", "Aura::HeartbeatResistance: Breaking creature aura %u. Seconds passed %u with chance %u.", m_spellInfo->Id, m_heartBeatTimer, breakPct);
+            }
+        }
+
+        m_heartBeatTimer = m_duration / 4;
```

Comment:
and here?

Reply:
yes

Reply:
basically you're checking at more and more closer periods of time instead of what's intended.
Suggestion, save the timer you calculate on construct (maxduration may be modified too)

Reply:
that depends on how a modified maxduration _should_ be handled (i'm not sure)